packages/aws-cdk-lib/aws-codebuild/lib/fleet.ts

```diff
@@ -138,12 +204,24 @@ export class Fleet extends Resource implements IFleet {
       throw new Error('baseCapacity must be greater than or equal to 1');
     }
 
+    if (props.computeConfiguration && props.computeType !== FleetComputeType.ATTRIBUTE_BASED) {
+      throw new Error('computeConfiguration can only be specified if computeType is "ATTRIBUTE_BASED"');
+    }
+
+    if (props.computeType === FleetComputeType.ATTRIBUTE_BASED &&
+      (!props.computeConfiguration || Object.keys(props.computeConfiguration).length === 0)) {
+      throw new Error('At least one compute configuration criteria must be specified if computeType is "ATTRIBUTE_BASED"');
+    }
```

Comment:
I'm only doing base prop validation here. I think currently invalid attribute configurations, such as the 192 GiB of memory and NVME machine type described in the integ test, shouldn't be validated by the CDK. These limitations are bound to change over time and across regions.